client/src/main/java/io/cloudsoft/winrm4j/client/WinRmClient.java

```diff
@@ -215,20 +285,99 @@ private WinRm getService() {
         }
     }
 
+    private class JaxWsClientWithNegotiateFactoryBean extends JaxWsClientFactoryBean {
+        @Override
+        protected Client createClient(Endpoint ep) {
+            return new ClientImpl(getBus(), ep, getConduitSelector());
+        }
+
+    }
+
     private synchronized WinRm createService() {
         if (winrm != null) return winrm;
 
         WinRmService service = new WinRmService();
+//        JaxWsDynamicClientFactory dcf = JaxWsDynamicClientFactory.newInstance();
+//        Client client = dcf.createClient("people.wsdl", classLoader);
         winrm = service.getWinRmPort(
                 // * Adds WS-Addressing headers and uses the submission spec namespace
                 //   http://schemas.xmlsoap.org/ws/2004/08/addressing
                 newMemberSubmissionAddressingFeature());
 
+        // Needed to be async according to http://cxf.apache.org/docs/asynchronous-client-http-transport.html
+//        Bus bus = BusFactory.getDefaultBus();
+//        bus.setProperty(AsyncHTTPConduit.USE_ASYNC, Boolean.TRUE);
+
+        Client client = ClientProxy.getClient(winrm);
+        ServiceInfo si = client.getEndpoint().getEndpointInfo().getService();
+        si.setProperty("soap.force.doclit.bare", true);
+        si.setProperty("soap.no.validate.parts", true);
+
         BindingProvider bp = (BindingProvider)winrm;
+
+        Map<String, Object> requestContext = bp.getRequestContext();
+        AddressingProperties maps = new AddressingProperties(VersionTransformer.Names200408.WSA_NAMESPACE_NAME);
+        requestContext.put(JAXWSAConstants.CLIENT_ADDRESSING_PROPERTIES, maps);
+
         bp.getRequestContext().put(BindingProvider.ENDPOINT_ADDRESS_PROPERTY, endpoint.toExternalForm());
-        if (username != null && password != null) {
-            bp.getRequestContext().put(BindingProvider.USERNAME_PROPERTY, username);
-            bp.getRequestContext().put(BindingProvider.PASSWORD_PROPERTY, password);
+
+        switch(authenticationScheme) {
+            case AuthSchemes.BASIC:
+                bp.getRequestContext().put(BindingProvider.USERNAME_PROPERTY, username);
+                bp.getRequestContext().put(BindingProvider.PASSWORD_PROPERTY, password);
+            case AuthSchemes.NTLM:
+                Credentials creds = new NTCredentials(username, password, null, null);
+
+                CredentialsProvider credsProvider = new BasicCredentialsProvider();
+                credsProvider.setCredentials(
```

Comment:
Still not seeing where `credsProvider` is referenced after this line? You are setting up the object here, but not using it afterwards.